### PR TITLE
Bug 2106803: mock call to /api/devfile in e2e

### PR DESCRIPTION
**Pull request by invincibleJai**
**Fixes:** https://bugzilla.redhat.com/show_bug.cgi?id=2106803

image

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts [hunk 1]
```diff
@@ -21,6 +22,155 @@ interface GithubRepo {
   folder: string;
 }
 
+const devFileResources: DevfileSuggestedResources = {
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts:25]: @invincibleJai Should we add this as an additional attribute to the gitImporotRepos list?

```tsx
interface GithubRepo {
  url: string;
  folder: string;
  devFileResources?: DevfileSuggestedResources;
}
```

File: frontend/packages/dev-console/integration-tests/support/pages/add-flow/git-page.ts [hunk 2]
```diff
@@ -42,6 +42,11 @@ export const gitPage = {
         statusCode: responses.packageResponse ? 200 : 404,
         body: responses.packageResponse,
       }).as('getPackage');
+
+      cy.intercept('POST', '/api/devfile/', {
+        statusCode: responses.packageResponse ? 200 : 404,
+        body: responses.devFileResources,
+      }).as('getDevfileResources');
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/support/pages/add-flow/git-page.ts:49]: You want return 404 if there is no devfile defined, right?

```suggestion

      cy.intercept('POST', '/api/devfile/', {
        statusCode: responses.devFileResources ? 200 : 404,
        body: responses.devFileResources,
      }).as('getDevfileResources');
```

File: frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts [hunk 3]
```diff
@@ -31,5 +181,12 @@ export function getResponseMocks(repo: GithubRepo) {
...
     // nothing, the file does not exist
   }
-  return { repoResponse: repoJson, contentsResponse: contentsJson, packageResponse: packageJson };
+  return {
+    repoResponse: repoJson,
+    contentsResponse: contentsJson,
+    packageResponse: packageJson,
+    ...(repo.url === 'https://github.com/nodeshift-starters/devfile-sample' && {
+      devFileResources,
+    }),
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts:190]: Then you can avoid the special case here:

```suggestion
    devFileResources: repo.devFileResources,
```

**Review by jerolimov — CHANGES_REQUESTED** [review]
Some ideas, please take a look. Also fine with the current implementation if you want keep that.

File: frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts (hunk 1 above)
Comment (invincibleJai) [reply frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts:25]: have updated it to load similar to  repo.json, content.json, and package.json.

File: frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts [hunk 4]
```diff
@@ -26,10 +26,21 @@ export function getResponseMocks(repo: GithubRepo) {
...
   }
-  return { repoResponse: repoJson, contentsResponse: contentsJson, packageResponse: packageJson };
+  return {
+    repoResponse: repoJson,
+    contentsResponse: contentsJson,
+    packageResponse: packageJson,
+    ...(devFileResources && {
+      devFileResources,
+    }),
+  };
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts:45]:
```suggestion

  let devFileResources = null;
  try {
    devFileResources = require(`./${repo.folder}/devFileResources.json`);
  } catch (err) {
    // nothing, the file does not exist
  }

  return {
    repoResponse: repoJson,
    contentsResponse: contentsJson,
    packageResponse: packageJson,
    devFileResources,
  };
```

File: frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts [hunk 5]
```diff
@@ -26,10 +26,21 @@ export function getResponseMocks(repo: GithubRepo) {
   const contentsJson = require(`./${repo.folder}/contents.json`);
 
   let packageJson = null;
+  let devFileResources = null;
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/testData/git-import/repos.ts:29]:
```suggestion
```

**Review by jerolimov — COMMENTED** [review]
Sorry, but isn't it easier to just check if there is a devFileResource file in the repo folder. I believe there is no need to check the repo.url upfront. That would make it easier to add another devfile example in the future, or? :man_shrugging:

**Comment by invincibleJai** [comment]
> Sorry, but isn't it easier to just check if there is a devFileResource file in the repo folder. I believe there is no need to check the repo.url upfront. That would make it easier to add another devfile example in the future, or? 🤷‍♂️

yeah i added a check to avoid checking for devfile for all other flows but make sense to check always over checking against one URI , just that devFile resources will differ depending on URI i.e if in future some test add a diff GH url for devfile as mock response is base on https://github.com/nodeshift-starters/devfile-sample

**Comment by jerolimov** [comment]
Got saw a similar error also in 4.10. So let us backport this to reduce the number of GitHub API calls

/cherry-pick release-4.11

**Pull request merged**